### DISCUSSION TITLE
2.0-beta

### DISCUSSION THREAD

**Discussion by liamcain** [liamcain/obsidian-calendar-plugin#173] [General]
## New

- Overflow Menu: a fully configurable hover menu that shows key metadata for a given day
- Added new, configurable "Sources" that can be displayed on the calendar or in the overflow menu
- Added drag 'n drop support
- Added support for monthly notes
- Toggle 'week numbers' command

## Improved
- View should load faster and update quicker

## Related Issues
This release hopes to solve the following issues:
solves #169 
solves #163 
solves #161
solves #131
solves #102 _(kinda)_
solves #77 
solves #39 - Using the zettel source
solves #35 - can show in Overflow Menu

<hr /><em>This discussion was created from the release <a href='https://github.com/liamcain/obsidian-calendar-plugin/releases/tag/'>2.0-beta</a>.</em>

**Comment by wenlzhang** [comment]
Great work! Really like the new interface for displaying various options.

1. When testing this version, when trying to create a new Weekly Note or Monthly Note from the calendar view, it pops up with a window saying "New Daily Note" instead of "New Weekly Note" or "New Monthly Note".
2. If one chooses to show various options On Calendar, all the corresponding dots would show below a certain day. Is it possible to configure to show certain options at different locations of a certain day, e.g. below, above, to the left and to the right of a certain day?
3. When for Links and Backlinks, if there are links and backlinks for a certain day, they would show differently, although they are configured with the same option "In Menu". Specifically, it would say "0 Backlinks" but not the same for links. See the figure below for more info.

<img width="229" alt="CleanShot 2021-04-12 at 12 00 32@2x" src="https://user-images.githubusercontent.com/44930227/114377639-2ba42500-9b87-11eb-8021-f123c5c57544.png">

Also, a trivial thing is that when hovering the mouse on a certain Month/Week/Day for the first time, it takes some time, maybe 1 to 2 seconds, to show the info. After that, if one moves the mouse to another day, it shows almost instantly. Therefore, is it possible to show the info faster when hovering the mouse on a certain day for the first time?

**Reply by liamcain** [reply]
Thanks for taking an early look!

1. Good catch! I just fixed this locally, will have it in the next beta version
2. You're referring to where the dots are positioned relative to the date? I hadn't thought about that but my feeling is that it would look pretty messy. Are you just worried about how to distinguish between the dots regardless of color?
3. I still need to make this more clear, but the menu always shows the first two elements as "Showcased" -- So in this case, "Links" is showcased and "backlinks" is not. I'll give this some thought.
4. There's a 750ms delay before the hover appears to avoid it getting in the way.

**Reply by wenlzhang** [reply]
2. Yes. For instance,
```
        Words
Links | Date | Backlinks
        Tasks
```
Indeed, if the dots are placed like above, it can be a bit messy... My feeling is that there would be too many dots below the Date if they are shown in the current manner, when, for example, the options above are all configured to show On Calendar. In this case, it can be not obvious which colour indicates which option.

On the other hand, maybe it is a bit too much to show all of them On Calendar. I am not quite sure about this either. Just thought it might be interesting to see the options shown On Calendar.

Nevertheless, if only one dot is used to indicate if there are Links and Backlinks (instead of having many dots indicating the actual number of Links and Backlinks), would this be slightly better when the dots are shown to the left and right of the Date? Specifically, if there are Backlinks for a Daily Note, there would be a dot indicating this; otherwise, there would be no dots for Backlinks.

3. Aha, I see how this works now. Options in the first row would look a bit different compared with other options.

**Reply by liamcain** [reply]
Played around with dots in different positions and I don't think it's the right approach, it gets ambiguous very quickly. "Too many dots" was the original reason for the Overflow menu in the first place. There is also a max number of dots per source currently, and I'm debating adding a max per day as well with a `+` if it exceeds that threshold.

<img width="289" alt="image" src="https://user-images.githubusercontent.com/693981/114392550-f12a9000-9b66-11eb-8668-9cb292df4c04.png">

> Nevertheless, if only one dot is used to indicate if there are Links and Backlinks

I think this makes sense, since this is how the Tasks source works currently, I just didn't add that option to all sources yet.

> 3. Aha, I see how this works now. Options in the first row would look a bit different compared with other options.

I think I will move "hidden" sources into their own section at the bottom to make this more obvious too. But ideally I want this UI to feel like an iPhone home screen with Widgets: drag and drop to arrange, and customize how detailed a particular source is.

**Reply by wenlzhang** [reply]
> I'm debating adding a max per day as well with a + if it exceeds that threshold.

This can be very helpful.

> drag and drop to arrange, and customize how detailed a particular source is.

Didn't realize that one can re-arrange the options. This is a very nice feature. 👍

**Comment by wenlzhang** [comment]
When adding two tasks on a Monthly Note, e.g. April 2021, there would be two dots for all months in the Calendar, although those Monthly notes do not exist.

<img width="113" alt="CleanShot 2021-04-12 at 14 52 55@2x" src="https://user-images.githubusercontent.com/44930227/114397624-2a7ef200-9b9f-11eb-926d-24ae41db3525.png">

**Reply by liamcain** [reply]
Great catch, fixed. Will be in the next beta release.

**Comment by wenlzhang** [comment]
Options in the first row do not align well with those in the second row. This can be seen in the following figures.
<img width="252" alt="CleanShot 2021-04-12 at 15 05 12@2x" src="https://user-images.githubusercontent.com/44930227/114399001-bc3b2f00-9ba0-11eb-8692-c819fda6c620.png">
Fig. 1

<img width="258" alt="CleanShot 2021-04-12 at 15 05 35@2x" src="https://user-images.githubusercontent.com/44930227/114399014-bfceb600-9ba0-11eb-83ec-10b708d19f3f.png">
Fig. 2


In Fig. 2, the dot in the second row seems to be more to the left compared with the first figure.

**Reply by liamcain** [reply]
Yeah, this is hard to make look aligned in all cases. For example, if I have backlinks showcased first:
<img width="291" alt="image" src="https://user-images.githubusercontent.com/693981/114399921-71ed8a00-9b6f-11eb-9a65-0679ecf453b0.png">

Here you can see how the elements are meant to be aligned, but because with some of the border boxes and having a mix of left-aligned and right-aligned elements, it doesn't always look perfect. I will play around with the padding some more. This might be a case of "make it SO unaligned that it doesn't look like it should be aligned."

**Reply by wenlzhang** [reply]
From the figure you shared, I realized that I do not have the number indicating how many tasks there are for a certain day because I am using a light theme. May I ask which variables to change in the custom theme to make this work with a light theme?

**Reply by avtfaciam** [reply]
> From the figure you shared, I realized that I do not have the number indicating how many tasks there are for a certain day because I am using a light theme. May I ask which variables to change in the custom theme to make this work with a light theme?

I'm having the same issue. Attached screenshot.

![Screen Shot 2021-04-12 at 09 34 44](https://user-images.githubusercontent.com/66343542/114403588-eece3300-9b72-11eb-90cb-6cebe366fd96.png)

Would really appreciate knowing the variables! I didn't see anything in the styles.css file.

**Reply by liamcain** [reply]
Sooo, apparently I tested it in every theme _except_ default light theme. My bad. I just replaced the `main.js` in the Release to fix the coloring. I recommend redownloading the latest and replacing it in the plugin folder to fix it. Sorry about that @enjolrs @wenlzhang!

**Reply by avtfaciam** [reply]
I'm actually using the California Coast theme - but that fixed it! Thank you!

**Reply by wenlzhang** [reply]
Thanks for the update! It's also working for me now.

**Comment by wenlzhang** [comment]
It seems that one needs to configure Calendar again after restarting Obsidian, e.g. which options to show On Calendar, if one wants to show week number, etc.

**Reply by liamcain** [reply]
I can reproduce the issue, thanks for the report!

**Reply by liamcain** [reply]
Should be fixed in `2.0.0-beta2`

**Comment by wenlzhang** [comment]
Sometimes, the options would appear repeatedly in the configuration page, as well as in the Calendar interface. This happened a few times for me, and I am not sure what could have caused this. Re-launching Obsidian would resolve this issue, then one needs to re-configure everything.

<img width="1159" alt="CleanShot 2021-04-13 at 18 17 02@2x" src="https://user-images.githubusercontent.com/44930227/114586416-eb2ad100-9c84-11eb-856f-a6537390993a.png">
<img width="47" alt="CleanShot 2021-04-13 at 18 17 22@2x" src="https://user-images.githubusercontent.com/44930227/114586428-efef8500-9c84-11eb-8934-8c1d4a1292af.png">
There are three tasks in Week 15, meaning that the option Tasks appears five times in the configuration page.

**Reply by liamcain** [reply]
Does that happen after a restart or after you've made a configuration change? I haven't seen that behavior so trying to track it down now.

**Reply by wenlzhang** [reply]
That happened after launching Obsidian for some time. Specifically, I encountered this issue yesterday, then I re-started Obsidian and re-configured the Calendar options, and it worked as before. After that, I did not make any configuration change on Calendar. Today, I encountered this issue again, and I repeated this procedure to get rid of this issue.

**Reply by wenlzhang** [reply]
Do let me know how I can help to further identify this issue. :)

**Reply by liamcain** [reply]
Sorry for the long wait, I just published `2.0.0-beta2` which should improve the settings saving behavior. Let me know if you're still experiencing this issue!

**Reply by wenlzhang** [reply]
Great! Just installed it. By the way, I experienced the repeated icon issue a few times in the past few days.

**Reply by wenlzhang** [reply]
After linking a future daily note that does not exist yet, it shows that there are no backlinks to that daily note when hovering on the corresponding date on the calendar. Is it possible to fix this?

**Reply by liamcain** [reply]
> After linking a future daily note that does not exist yet, it shows that there are no backlinks to that daily note when hovering on the corresponding date on the calendar. Is it possible to fix this?

yeah, this is know (the same thing happens with zettels). I am still figuring out the best way to handle this but I'm hoping to have it fixed before launch.

**Reply by liamcain** [reply]
> By the way, I experienced the repeated icon issue a few times in the past few days. 

If it happens again in the new version, could you send me a copy of the `data.json` file in your plugin folder? It will help me debug.

**Reply by wenlzhang** [reply]
Sure. I'll let you know how it goes.

**Comment by wenlzhang** [comment]
Today, I experienced the repeated icon issue again, and it seems that, in the past days, I experienced the issue only once, which is not as frequent when using the last version of Calendar.

Here is what is included in my `data.json`:

```json
{"shouldConfirmBeforeCreate":true,"localeOverride":"system-default","weekStart":"locale","showWeeklyNote":true,"sourceSettings":{"wordCount":{"color":"#ebcb8b","display":"menu","wordsPerDot":250},"tasks":{"color":"#d08770","display":"calendar-and-menu","maxIncompleteTaskDots":3},"links":{"color":"#a3be8c","display":"menu"},"backlinks":{"color":"#5e81ac","display":"menu"},"zettels":{"color":"#b48ead","display":"none"},"emoji-tags":{"display":"none"}}}
```

**Reply by liamcain** [reply]
Okay, I tracked down the duplicate sources issue, will be fixed for good in the next beta. I happens if you close the calendar view and reopen it via `Calendar: Open View`

**Reply by wenlzhang** [reply]
@liamcain Is there any update on this issue? I almost experience this issue every day now.

**Reply by liamcain** [reply]
Ahhh, sorry for taking so long to get back to you. I'm embarrassingly behind on github notifications (and plugin stuff in general 😞 ). 

I think I fixed this back in April and never cut a new release. Let me double check.

**Comment by ukengo** [comment]
My planig is installed, but the calendar itself is not displayed. And there is no Calendar: Open view command. Please tell me how to fix this.

**Comment by andrea9292** [comment]
Please release 2.0 version... Your plugin is so great!

**Comment by mattmaiorana** [comment]
Just realized there's a beta that isn't in the main Obsidian repository - exciting! No idea if this is still being developed, but a few specific ideas:

1. Make the hover box totally optional. I like the simplicity of the version in the Obsidian repository. I don't mind if it's there as an option, but I don't want a hover box myself.
2. Move the month dots to the right of the month instead of below. You may have tried this and decided there's not enough space, but I think it would look better.

Thanks!

**Comment by theokang21** [comment]
Bjr, Svp à la page de paramétrage de Calendar, les options WEECKLY, MONTHÈLY et YEARLY ont disparu chez moi. Est-ce normal? comment le fixer ?
Merci

**Comment by theokang21** [comment]
Bjr je ne parviens pas à installer mes modèles. Il dit tjrs que aucun modèle n'est trouvé... Comment régler cela ? Merci

**Comment by Drakemoor** [comment]
Hi,
The latest [beta.2](https://github.com/liamcain/obsidian-calendar-plugin/releases/tag/2.0.0-beta.2) does not save the the setting for 'Start week on'. I tried it in the sandbox vault and it is the same.
